common/encryption/encrypt.h

```diff
@@ -29,17 +29,17 @@
 #include "mbedtls/x509_crt.h"
 #include "mbedtls/error.h"
 
-std::vector<unsigned char*> encrypt_bytes(std::string model_data) {
+unsigned char** encrypt_bytes(std::string model_data) {

```

Comment:
May also be good to make this function `void` and have it also accept a `unsigned char* ciphertext` argument to which you'll copy the resulting ciphertext (including IV and tag). i.e. instead of using `iv_out_tag` on line 55 you'll use this argument.

In this case you'll need to allocate an array of the proper size (which should be known since you know the length of `model_data` as well as the `CIPHER_IV_SIZE` and `CIPHER_TAG_SIZE` and pass in this array as the `ciphertext` argument.